internal/pkg/cli/app_show_test.go

```diff
@@ -0,0 +1,331 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/cli/mocks"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+type showAppMocks struct {
+	storeSvc *mocks.Mockstore
+	prompt   *mocks.Mockprompter
+	sel      *mocks.MockappSelector
+}
+
+func TestShowAppOpts_Validate(t *testing.T) {
+	testError := errors.New("some error")
+	testCases := map[string]struct {
+		inAppName  string
+		setupMocks func(mocks showAppMocks)
+
+		wantedError error
+	}{
+		"valid app name": {
+			inAppName: "my-app",
+
+			setupMocks: func(m showAppMocks) {
+				m.storeSvc.EXPECT().GetApplication("my-app").Return(&config.Application{
+					Name: "my-app",
+				}, nil)
+			},
+			wantedError: nil,
+		},
+		"invalid app name": {
+			inAppName: "my-app",
+
+			setupMocks: func(m showAppMocks) {
+				m.storeSvc.EXPECT().GetApplication("my-app").Return(nil, testError)
+			},
+
+			wantedError: fmt.Errorf("get application %s: %w", "my-app", testError),
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStoreReader := mocks.NewMockstore(ctrl)
+			mockPrompter := mocks.NewMockprompter(ctrl)
+
+			mocks := showAppMocks{
+				storeSvc: mockStoreReader,
+				prompt:   mockPrompter,
+			}
+			tc.setupMocks(mocks)
+
+			showProjects := &showAppOpts{
```

Comment:
maybe showApp?

Reply:
fixed it here! https://github.com/aws/amazon-ecs-cli-v2/pull/927